.github/workflows/pathogen-repo-build.yaml.in

```diff
@@ -218,6 +229,33 @@ jobs:
           | "$NEXTSTRAIN_GITHUB_DIR"/bin/json-to-envvars
           | tee -a "$GITHUB_ENV"
 
+      - id: role
+        name: Set repo-specific role to (potentially) assume for runtime access to AWS
+        env:
+          REPO_FULL_NAME: ${{ github.repository }}
+        run: |
+          echo "arn=arn:aws:iam::827581582529:role/GitHubActionsRoleNextstrainRepo@${REPO_FULL_NAME#*/}" | tee -a "$GITHUB_OUTPUT"
+
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: us-east-1
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          role-to-assume: ${{ secrets.AWS_ACCESS_KEY_ID == '' && steps.role.outputs.arn || '' }}
+          role-duration-seconds: 43200 # seconds, or 12 hours
```

Comment:
We could potentially add `continue-on-error` here if there's a use case to run builds without AWS credentials, but that will probably cause more issues than it's worth...

Reply:
Yeah, I thought about that but wasn't sure what the use case would be. Do we have need to use pathogen-repo-build _without_ interacting with AWS at all (e.g. no download of private data, no upload of any kind)?

I guess if we were only downloading public data and uploading to Nextstrain Groups or similar. But I don't think we do that right now.

Reply:
Yeah, I was thinking of the CI builds for a split second, but we're going to move away from `pathogen-repo-builds` for those anyways. 

I think fine to make the AWS role required.

Reply:
Sounds good. We can always change our minds later! ;-)